src/Miunie.WindowsApp/Converters/TokenMaskConverter.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using Windows.UI.Xaml.Data;
+
+namespace Miunie.WindowsApp.Converters
+{
+    public class TokenMaskConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, string language)
+        {
+            if (value is string token)
+            {
+                if (token == "[NOT CONFIGURED]") return value;
+
+                value = $"{token.Substring(0, 5)} ****************************** {token.Substring(token.Length - 5)}";
```

Comment:
If we were to use a `Substring` here, we would get an `IndexOutOfRangeException` when the user types in a string that is less than 5 characters and hits "Apply Anyway".

We can use the `.Take(5)` and `.TakeLast(5)` from `System.Linq`.

> ℹ️ Note that it returns an `IEnumerable<char>` so we'd have to make a string from it.